### PR TITLE
feat(update-creds): Allow the user to update credentials using the new ProfileInfo APIs

### PR DESCRIPTION
**Pull request by zFernand0**
## Changelog Updates:
- Fix https://github.com/zowe/vscode-extension-for-zowe/issues/1646
- Fix https://github.com/zowe/vscode-extension-for-zowe/issues/1720
- Fix https://github.com/zowe/vscode-extension-for-zowe/issues/1228
- 5356286cfc33567b3541effc3e820454ccf877b4 fixes https://github.com/zowe/vscode-extension-for-zowe/issues/1698
- https://github.com/zowe/zowe-cli/pull/1344 fixes https://github.com/zowe/vscode-extension-for-zowe/issues/1697
- Added `Update Credentials` command to the command palette + Expose as right-click option in `098_zowe_<xyz>ProfileAuthentication` (Same group as SSO Login and Logout)

temp

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

## Additional Context
This PR depends on:
- https://github.com/zowe/imperative/pull/748 
- https://github.com/zowe/zowe-cli/pull/1344

### CODE REVIEW

File: packages/zowe-explorer-api/src/profiles/ProfilesCache.ts [hunk 1]
```diff
@@ -419,10 +419,13 @@ export class ProfilesCache {
      * not initialized it we mock a default value.
      */
     private getZoweDir(): string {
-        imperative.ImperativeConfig.instance.loadedConfig = {
-            defaultHome: path.join(os.homedir(), ".zowe"),
-            envVariablePrefix: "ZOWE",
-        };
+        if (imperative.ImperativeConfig.instance.loadedConfig == null) {
```
Comment (zFernand0) [inline packages/zowe-explorer-api/src/profiles/ProfilesCache.ts:422]: If there are concerns about setting loadedConfig only once (e.g. the `os.homedir()` may change from command to command ?) ...

We can replace 
```javascript
if (imperative.ImperativeConfig.instance.loadedConfig == null)
```
for
```javascript
if (imperative.ImperativeConfig.instance.loadedConfig?.defaultHome !== path.join(os.homedir(), ".zowe"))
```
---
A better approach might be to replace this private function with [`ProfileUtils.getZoweDir()`](https://github.com/zowe/zowe-cli/blob/next/packages/core/src/utils/ProfileUtils.ts#L20) and fix this same issue there. 😋

Comment (zFernand0) [reply packages/zowe-explorer-api/src/profiles/ProfilesCache.ts]: Removed private function and fixed the issue in https://github.com/zowe/zowe-cli/pull/1344

**Comment by zFernand0** [comment]
See checks for the previous run:
image

**Review by JillieBeanSim — APPROVED** [review]
LGTM thanks @zFernand0 
I don't see the error anymore with v1 profiles but do with the v2 without secure fields with `autostore: true`.  if `autostore: false` I don't see the issue.  We can look into a solution for a better error message or something in a follow-up PR

I am still getting the error with `alt+z` but found where it's coming from.  It is coming from the call to the KeytarApi in extension.ts and we have an existing issue we can use to track fixing that.

**Review by rudyflores — APPROVED** [review]
LGTM! Tested creating a v2 profile and updating the credentials to my profiles with wrong creds and seems to be updating as intended! Tests seemed to have run just fine as well. Thanks @zFernand0 !

**Pull request merged**